src/publisher.rs

```diff
@@ -310,17 +311,18 @@ impl<T: ShmSend> Publisher<[T]> {
             .try_allocate_slice(len as u32, align as u32)
             .ok_or(IceoryxError::LoanSampleFailed)?;
 
-        let data = unsafe {
+        let _data = unsafe {
             let data = from_raw_parts_mut(data as *mut MaybeUninit<T>, len as usize);
             Box::from_raw(data)
         };
```

Comment:
@phil-opp this is a first shot to introduce a raw sample. I wanted to hit two birds with one stone and also get rid of the `Box` API in the sys crate. It worked quite well until I came to this code with a boxed slice.

My tendency leans now towards a custom `NonNull` implementation for sys.

Regarding the issue with the boxed slice, do you have a better idea than this
```rust
RawSampleMut<T> {
    payload: *mut T,
    slice_len: usize,
}
```
When `T` isn't a slice `slice_len` would be 0.

Reply:
I haven't looked at the other code in detail, but is there a reason why we can't just set `T` to an unsized slice, i.e. `RawSampleMut<[u8]>`? Then we would not need an additional `slice_len` field at all.

Reply:
@phil-opp maybe my Rust knowledge lack here but when we use an unsized slice, internally the is a runtime information about the size of the slice. A `Box<[u8]>` also has a pointer and a size field instead of just a pointer. At least to my knowledge.